packages/editor/src/components/post-actions/actions.js

```diff
@@ -37,13 +36,123 @@ import { exportPatternAsJSONAction } from './export-pattern-action';
 // Patterns.
 const { PATTERN_TYPES } = unlock( patternsPrivateApis );
 
+/**
+ * Check if a template is removable.
+ *
+ * @param {Object} template The template entity to check.
+ * @return {boolean} Whether the template is removable.
+ */
+function isTemplateRemovable( template ) {
+	if ( ! template ) {
+		return false;
+	}
+	// In patterns list page we map the templates parts to a different object
+	// than the one returned from the endpoint. This is why we need to check for
+	// two props whether is custom or has a theme file.
+	return (
+		[ template.source, template.templatePart?.source ].includes(
+			TEMPLATE_ORIGINS.custom
+		) &&
+		! template.has_theme_file &&
+		! template.templatePart?.has_theme_file
```

Comment:
I don't understand why the template part object is actually template.templatePart and not just template?

Reply:
In patterns list page we map the templates parts to a different object than the one returned from the endpoint. This is why we need to check two props.

Reply:
Can we avoid mapping and just return the right objects?

Reply:
I encountered the same issue when fixing a bug for the "Reset" action - #62951. The fact that action might receive differently shaped objects when viewing a single or list of entities can be confusing.

Reply:
I'm fixing that in https://github.com/WordPress/gutenberg/pull/62927

Reply:
> Can we avoid mapping and just return the right objects?

Probably.. I hadn't implemented that and will check your PR @youknowriad